pyqtgraph/palette.py

```diff
@@ -0,0 +1,414 @@
+        'm8':'#dfdbd8', 'm9':'#fdf8f5',
+        # --- functional colors ---
+        'gr_fg' : 'm5', 'gr_bg' : 'm9', #'#101518',
+        'gr_txt': 'm2', 'gr_acc': '#ad5a00', 
+        'gr_hlt': '#0080ff', 'gr_reg': ('#b36b1e',160),
+        # legacy colors:
+        'b':'p7', 'c':'p0', 'g':'#509d46', 'y':'p1', 'r':'p3' ,'m':'p5',
+        'k':'m0', 'd':'m3', 'l':'m6', 'w': 'm9', 
+        's': 'gr_hlt'
+    },
+    'pastels':{
+        'colormap_sampling': ('CET-C7', 0.060, +0.125),
+        # --- slightly warm (CIElab A=1 B=2) monochrome ramp ---
+        'm0':'#100c08', 'm1':'#262321',
+        'm2':'#3d3a37', 'm3':'#55524f',
+        'm4':'#6f6b68', 'm5':'#8a8683',
+        'm6':'#a5a19e', 'm7':'#c2bdba',
+        'm8':'#dfdbd8', 'm9':'#fdf8f5',
+        # --- functional colors ---
+        'gr_fg' : 'm6', 'gr_bg' : 'm9', #'#101518',
+        'gr_txt': 'm3', 'gr_acc': '#e07050', 
+        'gr_hlt': '#e03020', 'gr_reg': ('#ffc0b0',160),
+        # legacy colors:
+        'b':'p3', 'c':'p4', 'g':'p5', 'y':'p6', 'r':'p1' ,'m':'p2',
+        'k':'m0', 'd':'m3', 'l':'m6', 'w': 'm9', 
+        's': 'gr_hlt'
+    },
+    'synthwave':{
+        'colormap_sampling': ('CET-L8', 0.275, 0.100),
+        # --- cool monochrome ramp with no true black ---
+        'm0':'#16212a', 'm1':'#27353e',
+        'm2':'#3c4a54', 'm3':'#51606a',
+        'm4':'#687782', 'm5':'#808f9a',
+        'm6':'#98a8b3', 'm7':'#b1c1cd',
+        'm8':'#cbdce7', 'm9':'#e6f2ff',
+        # --- functional colors ---
+        'gr_fg' : '#599FA6', 'gr_bg' : 'm0',
+        'gr_txt': '#00E0FF', 'gr_acc': '#40B0BF',
+        'gr_hlt': '#00E0FF', 'gr_reg': ('#599CA6',170),
+        # legacy colors:
+        'b':'#398bfc', 'c':'gr_txt', 'g':'#39fc49', 'y':'p7', 'r':'p4' , 'm':'p1',
+        'k':'m0' , 'd': 'm3', 'l': 'm6', 'w':'m9', 's': 'gr_hlt'
+    }
+}
+
+def identifier_to_QColor( identifier, color_dict=None ):
+    """ 
+    Convert color information to a QColor 
+    ===================  =============================================================
+    **allowed formats**    
+    'name'               name must be a hex value or a key in 'color_dict'
+    ('name', alpha)      new copy will be assigned the specified alpha value
+    QColor               will be copied to avoid interaction if changing alpha values
+    Qt.GlobalColor       will result in a matching QColor 
+    (R,G,B)              a new Qcolor will be created
+    (R,G,B, alpha)       a new Qcolor with specified opacity will be created
+    ===================  =============================================================
+    """
+    if isinstance(identifier, (QtGui.QColor, QtCore.Qt.GlobalColor)):
+        return QtGui.QColor(identifier)
+    alpha = None
+    if isinstance(identifier, str): # return known QColor
+        name = identifier
+        if color_dict is None or name not in color_dict:
+            if name[0] != '#':
+                raise ValueError('Undefined color name '+str(identifier))
```

Comment:
would again recommend a f-string here `f"Undefined color name {identifier}"`

Reply:
I will grudgingly admit that these seem quite useful. --> changed